crawlers/moonstreamcrawlers/ethereum.py

```diff
@@ -105,3 +110,49 @@ def crawl(block_numbers_list: List[int], with_transactions: bool = False) -> Non
                 block_numbers_list[worker - 1 :: MOONSTREAM_CRAWL_WORKERS],
                 with_transactions,
             )
+
+
+def process_contract_deployments() -> List[Tuple[str, str]]:
+    """
+    Checks for new smart contracts that have been deployed to the blockchain but not registered in
+    the smart contract registry.
+
+    If it finds any such smart contracts, it retrieves their addresses from the transaction receipts
+    and registers them in the smart contract registry.
+
+    Returns a list of pairs of the form [..., ("<transaction_hash>", "<contract_address>"), ...].
+    """
+    web3_client = connect()
+    results: List[Tuple[str, str]] = []
+    with yield_db_session_ctx() as db_session:
+        current_offset = 0
+        limit = 10
+        transactions_remaining = True
+        while transactions_remaining:
+            contract_deployments = (
+                db_session.query(EthereumTransaction)
```

Comment:
This query should only pick transactions which aren't already represented in the `ethereum_smart_contracts` table.